pkg/kubelet/kubelet.go

```diff
@@ -526,13 +526,13 @@ func NewMainKubelet(kubeCfg *kubeletconfiginternal.KubeletConfiguration,
 		nodeIP:               parsedNodeIP,
 		nodeIPValidator:      validateNodeIP,
 		clock:                clock.RealClock{},
-		enableControllerAttachDetach:            kubeCfg.EnableControllerAttachDetach,
-		iptClient:                               utilipt.New(utilexec.New(), utildbus.New(), utilipt.ProtocolIpv4),
-		makeIPTablesUtilChains:                  kubeCfg.MakeIPTablesUtilChains,
-		iptablesMasqueradeBit:                   int(kubeCfg.IPTablesMasqueradeBit),
-		iptablesDropBit:                         int(kubeCfg.IPTablesDropBit),
-		experimentalHostUserNamespaceDefaulting: utilfeature.DefaultFeatureGate.Enabled(features.ExperimentalHostUserNamespaceDefaultingGate),
-		keepTerminatedPodVolumes:                keepTerminatedPodVolumes,
+		enableControllerAttachDetach: kubeCfg.EnableControllerAttachDetach,
+		iptClient:                    utilipt.New(utilexec.New(), utildbus.New(), utilipt.ProtocolIpv4),
+		makeIPTablesUtilChains:       kubeCfg.MakeIPTablesUtilChains,
+		iptablesMasqueradeBit:        int(kubeCfg.IPTablesMasqueradeBit),
+		iptablesDropBit:              int(kubeCfg.IPTablesDropBit),
+		nodeUserNamespace:            utilfeature.DefaultFeatureGate.Enabled(features.NodeUserNamespace),
```

Comment:
I find this confusing. Is this feature gated by `--feature-gates=NodeUserNamespace=true` or a separate flag?

Reply:
Also, will this be set to false if the feature gate is disabled?

Reply:
yes, this can be enabled by `--feature-gates=NodeUserNamespace=true` and will be set to false if feature gate is disabled.
Please suggest how can make it more clear.

Reply:
My bad. I thought the nodeUserNamespace was populated by a separate flag. Did not that both the old and the new features were controlled by gates.

Does this mean that the beta feature `ExperimentalHostUserNamespaceDefaultingGate` no longer works?

Reply:
`ExperimentalHostUserNamespaceDefaultingGate` was already broken since 1.7, after migration to dockershim. Another confusing part is that though it was beta but still disabled by default.